naga/src/proc/typifier.rs

```diff
@@ -635,210 +644,46 @@ impl<'a> ResolveContext<'a> {
                 arg2: _,
                 arg3: _,
             } => {
-                use crate::MathFunction as Mf;
+                use crate::proc::OverloadSet as _;
+
+                let mut overloads = fun.overloads();
+                log::debug!("initial overloads for {:?}", fun);
+                log::debug!("{:#?}", overloads.as_debug(types));
```

Comment:
nitpick: You can coalesce these `log::debug!(…)` calls:

```suggestion
                log::debug!("initial overloads for {fun:?}\n{:#?}", overloads.as_debug(types));
```

In general, it's preferable to coalesce content that is intended to be emitted as a unit to a single call, because `log` backends might not necessarily be a simple line-separated output.

Reply:
This feedback also applies to other places where `log` is called in series.